frontend/src/app/services/scan.service.ts

```diff
@@ -71,6 +71,9 @@ export class ScanService {
     }
 
     public getRandomScan(taskKey: string): Promise<ScanMetadata> {
+        if (taskKey == null) {
```

Comment:
Currently we are mainly using `undefined` as empty value. Maybe we use it here too? Also compare with `===` operator.

Reply:
`params.get('task')` returns either string or null as I saw in the documentation. So I assumed that It would be better to compare it to null. But as far as I know `null == undefined` so I can change it undefined then.

Reply:
Aye thats true. Lets compare it to `undefined` to make it consistent.